geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreFactoryImpl.java

```diff
@@ -165,6 +168,10 @@ public DiskStore create(String name) {
           creation.addDiskStore(result);
         }
       }
+    } finally {
+      if (this.cache instanceof GemFireCacheImpl) {
+        ((GemFireCacheImpl) this.cache).unlockDiskStore(name);
+      }
```

Comment:
Can you the methods `lockDiskStore` and `unlockDiskStore` to the `InternalCache` interface instead?. It's cleaner as we won't need to cast to `GemFireCacheImpl`, besides that, we could use these methods in the future from other parts of the code.

Reply:
If `lockDiskStore`and `unlockDiskStore` are moved to `InternalCache`, all the classes that implement the interface has to implement `lockDiskStore`and `unlockDiskStore`. However, only `GemFireCacheImpl` actually needs `lockDiskStore`and `unlockDiskStore`. I don't think `CacheCreation` and `InternalCacheForClientAccess` need them.

Reply:
OK. Let's use the `default` implementation in `InternalCache`.